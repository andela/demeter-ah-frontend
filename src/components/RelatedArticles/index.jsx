import React from 'react';
import { Link } from 'react-router-dom';

const RelatedArticles = ({ articles, relatedArticleImg, description }) => (
  articles && articles.rows && articles.rows.length >= 1 ? (
    articles.rows.map(showArticle => (
      <div className="w-6/12 md:rel related bg-white ml-5" key={showArticle.slug}>
        <Link to={`/articles/${showArticle.slug}`}>
          <div className="cursor-pointer ">
            <div className="flex-1" style={relatedArticleImg(showArticle.image || '/placeholder.png')}> </div>
            <h2 className="title sm:min-w-12 lg:min-w-102 sm:text-xs-2 md:text-xl lg:text-2xl">{showArticle.title}</h2>
            {!description ? '' : <p className="pb-4">{description || showArticle.description}</p>}
          </div>
        </Link>
      </div>
    ))
  ) : (
    <p className="text-xs text-center">No related articles yet</p>
  )
);

export default RelatedArticles;
