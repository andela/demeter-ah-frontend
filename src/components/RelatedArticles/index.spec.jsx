import React from 'react';
import { shallow } from 'enzyme';
import RelatedArticles from './index';

describe('RelatedArticles Component', () => {
  it('Should render without errors', () => {
    const component = shallow(<RelatedArticles />);
    expect(component).toMatchSnapshot();
  });
});
