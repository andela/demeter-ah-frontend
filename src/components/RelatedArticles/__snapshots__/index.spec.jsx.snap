// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RelatedArticles Component Should render without errors 1`] = `
<p
  className="text-xs text-center"
>
  No related articles yet
</p>
`;
